apps/web/modules/survey/follow-ups/lib/follow-ups.ts

```diff
@@ -0,0 +1,266 @@
+        response,
+        attachResponseData: properties.attachResponseData,
+        logoUrl,
+      });
+
+      return {
+        followUpId: followUp.id,
+        status: "success" as const,
+      };
+    } catch (error) {
+      return {
+        followUpId: followUp.id,
+        status: "error",
+        error: error instanceof Error ? error.message : "Something went wrong",
+      };
+    }
+  }
+
+  // If not a direct email, check if it's a question ID or hidden field ID
+  if (!toValueFromResponse) {
+    return {
+      followUpId: followUp.id,
+      status: "error",
+      error: `To value not found in response data for followup: ${followUp.id}`,
+    };
+  }
+
+  if (typeof toValueFromResponse === "string") {
+    // parse this string to check for an email:
+    const parsedResult = z.string().email().safeParse(toValueFromResponse);
+    if (parsedResult.data) {
+      // send email to this email address
+      await sendFollowUpEmail({
+        followUp,
+        to: parsedResult.data,
+        replyTo,
+        logoUrl,
+        survey,
+        response,
+        attachResponseData: properties.attachResponseData,
+      });
+
+      return {
+        followUpId: followUp.id,
+        status: "success" as const,
+      };
+    } else {
+      return {
+        followUpId: followUp.id,
+        status: "error",
+        error: `Email address is not valid for followup: ${followUp.id}`,
+      };
+    }
+  } else if (Array.isArray(toValueFromResponse)) {
+    const emailAddress = toValueFromResponse[2];
+    if (!emailAddress) {
+      return {
+        followUpId: followUp.id,
+        status: "error",
+        error: `Email address not found in response data for followup: ${followUp.id}`,
+      };
+    }
+
+    const parsedResult = z.string().email().safeParse(emailAddress);
+    if (parsedResult.data) {
+      await sendFollowUpEmail({
+        followUp,
+        to: parsedResult.data,
+        replyTo,
+        logoUrl,
+        survey,
+        response,
+        attachResponseData: properties.attachResponseData,
+      });
+
+      return {
+        followUpId: followUp.id,
+        status: "success" as const,
+      };
+    } else {
+      return {
+        followUpId: followUp.id,
+        status: "error",
+        error: `Email address is not valid for followup: ${followUp.id}`,
+      };
+    }
+  }
+
+  return {
+    followUpId: followUp.id,
+    status: "error",
+    error: "Something went wrong",
+  };
+};
+
+/**
+ * Sends follow-up emails for a survey response.
+ * This is the main entry point for sending follow-ups - it handles all the logic internally
+ * and only requires a response ID.
+ */
+export const sendFollowUpsForResponse = async (
+  responseId: string
+): Promise<Result<FollowUpResult[], { code: FollowUpSendError; message: string; meta?: any }>> => {
+  try {
+    validateInputs([responseId, ZId]);
+    // Get the response first to get the survey ID
+    const response = await getResponse(responseId);
```

Comment:
We are still doing three database/cache calls (getResponse, getSurvey, getOrganization) where we could only do one. But I think we can also solve this in a future refactor.